active/AnimatedIcon/AnimatedIconSpec.md

```diff
@@ -0,0 +1,894 @@
+and the `Markers` property to find named positions in the animation timeline.
+
+_Spec note: `IAnimatedVisualSource2` is like `IAnimatedVisualSource` in purpose and use,
+but there is no derivation because the '2' interface does not derive from
+[IDynamicAnimatedVisualSource](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.IDynamicAnimatedVisualSource)._
+
+This example displays and animates an object implementing IAnimatedVisualSource2:
+
+```cpp
+void AddVisualAndShowStartAnimation(
+    const winrt::Border& element,
+    const winrt::IAnimatedVisualSource2 source,
+    const winrt::hstring& initialState,
+    const winrt::hstring& steadyState,
+    const winrt::Color& themeColor )
+{
+    winrt::IAnimatedVisual animatedVisual = source.TryCreateAnimatedIconVisual();
+
+    // Find the positions in the animation of the two states
+    auto const markers = source.Markers())
+    auto const fromProgress = static_cast<float>(markers.Lookup(initialState));
+    auto const toProgress = static_cast<float>(markers.Lookup(steadyState));
+
+    // Set the theme color as the animated visual's foreground.
+    source.SetColorProperty("Foreground", themeColor);
+
+    // Helper that uses TryCreateAnimatedVisual to add the source to the XAML element tree.
+    // See IAnimatedVisualSource.TryCreateAnimatedVisual for more information.
+    AddVisualToElement(element, animatedVisual);
+
+    // Helper to play an animation.
+    // See IAnimatedVisual for more information.
+    PlaySegment(animatedVisual.RootVisual(), fromProgress, toProgress);
+}
+```
+
+## Other IAnimatedVisualSource2 interface members
+
+| Name | Description |
+| :-| :-|
+| Markers | Provides a mapping from marker names to positions in the animation. The names are defined by the animation. |
+| SetColorProperty | Sets a color for the animated visual. The property names are defined by the animation. |
+
+## AnimatedAcceptVisualSource class
+
+This class can be used to show an animation for an Accept operation.
+This can be used as the Source of an AnimatedIcon and is used by the XAML CheckBox control.
+
+This class supports markers that can be driven by an AnimatedIcon.State:
+* NormalOnToPointerOverOn
+* NormalOnToPressedOn
+* NormalOffToNormalOn
+* NormalOffToPointerOverOff
+* NormalOffToPressedOff
+* PointerOverOnToPointerOverOff
+* PointerOverOnToNormalOn
+* PointerOverOnToPressedOn
+* PointerOverOffToPointerOverOn
+* PointerOverOffToNormalOff
+* PointerOverOffToPressedOff
+* PressedOnToPressedOff
+* PressedOnToPointerOverOff
+* PressedOnToNormalOff_Start
+* PressedOffToPressedOn
+* PressedOffToPointerOver
+* PressedOffToNormalOn
+
+## AnimatedGlobalNavigationButtonVisualSource class
+
+This class can be used to show an animation for a Navigation operation.
+This can be used as the Source of an AnimatedIcon and is used by the XAML NavigationView control.
+
+This class supports markers that can be driven by an AnimatedIcon.State:
```

Comment:
Again, we should be documenting states, not markers. The algorithm for generating markers from states is already described elsewhere, and it avoids having to provide a list of the combinatorial explosion of markers. Saves us a bunch of typing, and saves the developer a bunch of reading (and then trying to reverse-engineer the states from the list of markers, like I had to do with the checkbox).

Reply:
Agree with @oldnewthing, we shouldn't list the markers but rather the states to save us and developers time.